### ISSUE TITLE
intersight_vnic_eth_adapter_policy not idempotent when roce_settings disabled

### ISSUE THREAD

**Issue by gve-vse-tim** [CiscoDevNet/terraform-provider-intersight#62]
In the resource 'intersight_vnic_eth_adapter_policy', when roce_settings->enabled = false, you should not have to define any of the remaining block arguments (memory_regions, queue_pairs, and resource_groups).  However, if you don't define them, the resource is not idempotent:

```
# Excerpt of terraform configuration
resource "intersight_vnic_eth_adapter_policy" "default_ethernet" {
    organization {
        moid = data.intersight_organization_organization.default.moid
    }

    description = "Terraform deployed"
    name = "default_ethernet"

    roce_settings {
        enabled = false
    }
}

# Excerpt if output of terraform plan *after* the first terraform apply
  # intersight_vnic_eth_adapter_policy.default_ethernet will be updated in-place
  ~ resource "intersight_vnic_eth_adapter_policy" "default_ethernet" {
        id                        = "604530ad02bcdf21a9781b59"
        name                      = "default_ethernet"
      ~ roce_settings             = [
          ~ {
              ~ memory_regions        = 0 -> 131072
              ~ queue_pairs           = 0 -> 256
              ~ resource_groups       = 0 -> 4
                # (6 unchanged elements hidden)
            },
        ]
        # (19 unchanged attributes hidden)
    }

```

Please consider changing the provider to ignore those attributes (or update the local state with the remote default values) when 'enabled = false'.

Thanks!

**Comment by gve-vse-tim** [comment]
Update - the situation is slightly different (worse?).  It seems with roce_settings->enabled = false, those 3 attributes are forced to zero.  When I try to set those values - even with enabled = false - in order to provide a workaround to achieve idempotency, the 0 values persist:

```
# Excerpt of terraform configuration
resource "intersight_vnic_eth_adapter_policy" "default_ethernet" {
    organization {
        moid = data.intersight_organization_organization.default.moid
    }

    description = "Terraform deployed"
    name = "default_ethernet"

    roce_settings {
        enabled = false
        memory_regions = 131072
        queue_pairs = 256
        resource_groups = 4
    }

# Excerpt if output of terraform plan *after* the terraform apply
  # intersight_vnic_eth_adapter_policy.default_ethernet will be updated in-place
  ~ resource "intersight_vnic_eth_adapter_policy" "default_ethernet" {
        id                        = "604530ad02bcdf21a9781b59"
        name                      = "default_ethernet"
      ~ roce_settings             = [
          ~ {
              ~ memory_regions        = 0 -> 131072
              ~ queue_pairs           = 0 -> 256
              ~ resource_groups       = 0 -> 4
                # (6 unchanged elements hidden)
            },
        ]
        # (19 unchanged attributes hidden)
    }

```

So, it would seem that the provider code was set up to ignore the values if provided (?) but the Intersight API clearly doesn't like 0's for those values.

**Comment by code-lucidal58** [comment]
When `enabled` of `roce_settings` is set to `false`, the properties `memory_regions`, `queue_pairs` and `resource_groups` should be set to 0 explicitly. This will clear any settings in the endpoint. The other set of values i.e. (131072, 256, 4) are applicable only when `enabled` is set to `true`. I understand Intersight UI behaves differently in the sense that it does not allow users to set the latter 3 properties when the `enabled` radio button is off. If you look from the endpoint side, the 3 properties always have some value and they are always referred to, for other computation, even when `enabled` if false. That is why, they should always bear a valid value. 
Keeping this in mind, the following configurations will not show any drifts when `terraform plan` is executed after `terraform apply`.
**Enabled in RoceSettings is False**
```
resource "intersight_vnic_eth_adapter_policy" "default_ethernet" {
    organization {
        moid = data.intersight_organization_organization.default.results.0.moid
    }
    description = "Terraform deployed"
    name = "default_ethernet"
    roce_settings {
        enabled = false
        memory_regions = 0
        queue_pairs = 0
        resource_groups = 0
    }
}
```
**Enabled in RoceSettings is True. Other properties inside RoceSettings will pick up default non-zero values**
```
resource "intersight_vnic_eth_adapter_policy" "default_ethernet" {
    organization {
        moid = data.intersight_organization_organization.default.results.0.moid
    }
    description = "Terraform deployed"
    name = "default_ethernet"
    roce_settings {
        enabled = true
    }
}

**Comment by meghana-rajashekar** [comment]
#### Solved in the latest provider release 1.0.23 
Provider now ignores the roce_settings attributes when it's disabled (enabled = false) using the DiffSuppressFunc()
